src/towncrier/_settings/load.py

```diff
@@ -7,19 +7,49 @@
 import sys
 
 from collections import OrderedDict
-from typing import Any, Mapping
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, Mapping
 
 import pkg_resources
 
 from .._settings import fragment_types as ft
 
 
+if TYPE_CHECKING:
+    # We only use Literal for type-checking and Mypy always brings its own
+    # typing_extensions so this is safe without further dependencies.
```

Comment:
I don't understand this comment.  What is always safe?  And what does it mean to put a runtime check like `sys.version_info < (3, 8)` inside an `if TYPE_CHECKING:` guard (which guarantees it will never run).  Does mypy know what `sys.version_info < (3, 8)` means?

Reply:
It's safe to import it from typing_extensions, because when there's a type-checker, there should be typing_extensions.

Yes it does, unlike ImportErrors: https://hynek.me/til/typing-protocol/

I could probably always import from typing_extensions, but I like to have it explicit like this, so we can kill those branches wholesale once we drop 3.7.